includes/PhysicsEngine.h

```diff
@@ -46,4 +46,25 @@ class ForwardEuler : public PhysicsEngine {
     std::shared_ptr<spdlog::logger> euler_logger;
 };
 
+struct State {
+    Vector3 vel;
+    Vector3 accel;
+    Vector3 ang_vel;
+    Vector3 ang_accel;
+};
```

Comment:
The name `Struct` is quite generic, and I can foresee it conflicting with other software components in the future.

Since this struct is specific to your RK4 algorithm, you should define this struct in the `private:` sector of the `RungeKutta` class. That way the name `State` is only relevant within the class implementation.

While you're at it, I'd also suggest renaming this to `RungeKuttaState` for the sake of verbosity, but that's up to you